dev/NavigationView/NavigationView.cpp

```diff
@@ -2640,6 +2640,14 @@ void NavigationView::OnPropertyChanged(const winrt::DependencyPropertyChangedEve
     {
         UpdatePaneVisibility();
         UpdateVisualStateForDisplayModeGroup(DisplayMode());
+
+        if (auto splitView = m_rootSplitView.get())
```

Comment:
It's possible that just a timing issue. PaneOpen/PaneClose has animation and it's not a sync call.
Also we have a two way binding NavView.IsPaneOpen <-> SplitView.IsPaneOpen.
First we can try wait in the test case. 
GetToggledWaiter
RetryUntilEvalFuncSuccessOrTimeout
Also It's possible that when test case fail, try to read the status of NavView.IsPaneOpen and checkbox state

Reply:
There is a real product bug that has been reported though -- we just don't have a live repro of it yet. It's possible that this test has actually hit it, which would be great! See [this bug report](https://microsoft.visualstudio.com/OS/_workitems/edit/20525786).

Reply:
I managed to repro it, updated pr description with repro steps.